pkg/kwok/metrics/metrics.go

```diff
@@ -74,10 +71,6 @@ func NewMetricsUpdateHandler(conf UpdateHandlerConfig) (*UpdateHandler, error) {
 		registry:    registry,
 		handler:     handler,
 	}
-	err := h.init()
-	if err != nil {
-		return nil, fmt.Errorf("failed to initialize metrics update handler: %w", err)
-	}
 	return h, nil
```

Comment:
Since this can't return an error, we can eliminate it from the return type and just return `&UpdateHandler{...}`